src/pages/index.js

```diff
@@ -52,36 +138,91 @@ function createCardObject() {
-    popupEditProfile.close();
-    popupEditProfile.formElement.reset();
+    return api
+      .updateUserInfo({ userName: formData.name, about: formData.job })
+      .then((updatedUserData) => {
+        userInfo.setUserInfo({
+          name: updatedUserData.name,
+          job: updatedUserData.about,
+          avatar: updatedUserData.avatar,
+        });
+      })
+      .catch((err) => {
+        console.error(`Failed to update user data: ${err}`);
+      });
   },
 });
 
-// Popup for adding new cards
+// Popup for adding new cards old
+// const popupAddCard = new PopupWithForm({
+//   popupSelector: "#modal_adding-cards",
+//   handleFormSubmit: (formData) => {
+//     const cardData = { name: formData.title, link: formData.url };
+//     renderCard(cardData);
+//     cardFormValidator.disableBtn();
+//     popupAddCard.close();
+//     popupAddCard.formElement.reset();
+//   },
+// });
+
 const popupAddCard = new PopupWithForm({
   popupSelector: "#modal_adding-cards",
   handleFormSubmit: (formData) => {
-    const cardData = { name: formData.title, link: formData.url };
-    renderCard(cardData);
-    cardFormValidator.disableBtn();
-    popupAddCard.close();
-    popupAddCard.formElement.reset();
+    return api
+      .addNewCard({ locationName: formData.title, link: formData.url })
+      .then((newCardData) => {
+        renderCard(newCardData); // Add the newly created card to the UI
+        cardFormValidator.disableBtn();
+      })
+      .catch((err) => {
+        console.error(`Failed to add a new card: ${err}`);
+      });
   },
 });
 
+function handleLikeClick(cardData, card) {
+  if (cardData.isLiked === false) {
+    api.likeCard(cardData._id).then((newCardData) => {
+      card.updateLikeBtn(newCardData.isLiked);
+    });
+  } else {
+    api.unlikeCard(cardData._id).then((newCardData) => {
+      card.updateLikeBtn(newCardData.isLiked);
+    });
+  }
+}
+
 // Create and return a new card element
 function createCard(item) {
-  const cardElement = new Card(item, handleImageClick, "#card-template");
+  const cardElement = new Card(
+    item,
+    handleImageClick,
+    "#card-template",
+    handleDeleteClick,
+    handleLikeClick
+  );
   return cardElement.getView();
 }
 
-// Event listener for profile edit button
+// Event listener for profile edit button old code
+// profileEditBtn.addEventListener("click", () => {
+//   const userData = userInfo.getUserInfo();
+//   profileNameInput.value = userData.name;
+//   profileJobInput.value = userData.job;
+//   popupEditProfile.open();
+// });
+
 profileEditBtn.addEventListener("click", () => {
-  const userData = userInfo.getUserInfo();
-  profileNameInput.value = userData.name;
-  profileJobInput.value = userData.job;
-  popupEditProfile.open();
+  api
+    .getUserInfo()
+    .then((userData) => {
+      profileNameInput.value = userData.name;
+      profileJobInput.value = userData.about;
+      popupEditProfile.open();
+    })
+    .catch((err) => {
```

Comment:
Please, don't call `api.getUserInfo()` when you open the profile modal because you have everything when you call `userInfo.getUserInfo();`. The `UserInfo` class has everything you need


```js

profileEditBtn.addEventListener("click", () => {
  const userData = userInfo.getUserInfo();
  profileNameInput.value = userData.name;
  profileJobInput.value = userData.job;
  popupEditProfile.open();
});


```